hm_pyhelper/lock_singleton.py

```diff
@@ -11,17 +11,30 @@
 class LockSingleton(object):
     _prefix = "LockSingleton."
     _mode = 0o644
+    _default_resource_count = 1
 
-    def __init__(self, name, initial_value=1):
+    def __init__(self, name, resource_count=None):
+        """
+        name: uniquely identifies the `LockSingleton` in system wide.
+        resource_count: explicitly set the resource count.
+        """
         self._name = self._prefix + name
-        self._initial_value = initial_value
+        self._resource_count = resource_count or self._default_resource_count
         # so it doesn't interfere with our semaphore mode
         old_umask = os.umask(0)
         try:
             self._sem = posix_ipc.Semaphore(self._name,
                                             mode=self._mode,
                                             flags=posix_ipc.O_CREAT,
-                                            initial_value=initial_value)
+                                            initial_value=self._resource_count)
+            # Override the resource count if it's explicitly set. Otherwise the previous value
+            # of the resource count is preserved.
+            if resource_count and resource_count != self._sem.value:
```

Comment:
This code smells funny to me. Did you find any documentation online that say this is a common pattern?

Reply:
What value do we even get in wrapping `posix_ipc.Semaphore`? Why doesn't ecc_lock just use `posix_ipc.Semaphore` directly?

Reply:
Make sense. I refactored the this method to be easier to understand and added the comments how to use.